src/careamics/lightning/lightning_module.py

```diff
@@ -711,12 +705,106 @@ def create_careamics_module(
 
         # if use N2V
         if isinstance(algorithm_cfg, N2VAlgorithm):
-            algorithm_cfg.n2v_config.struct_mask_axis = struct_n2v_axis
-            algorithm_cfg.n2v_config.struct_mask_span = struct_n2v_span
+            algorithm_cfg.n2v_config.struct_mask_axis = (
+                algorithm_parameters["struct_mask_axis"]
+                if algorithm_parameters
+                else "none"
+            )
+            algorithm_cfg.n2v_config.struct_mask_span = (
+                algorithm_parameters["struct_n2v_span"] if algorithm_parameters else 5
+            )
+            use_n2v2 = (
+                algorithm_parameters["use_n2v2"] if algorithm_parameters else False
+            )
             algorithm_cfg.set_n2v2(use_n2v2)
 
         return FCNModule(algorithm_cfg)
     else:
         raise NotImplementedError(
             f"Algorithm {which_algo} is not implemented or unknown."
         )
+
+
+def create_vae_based_module(
+    algorithm: Union[SupportedAlgorithm, str],
+    loss: Union[SupportedLoss, str],
+    architecture: Union[SupportedArchitecture, str],
+    algorithm_parameters: dict | None = None,
+    model_parameters: dict | None = None,
+    optimizer: Union[SupportedOptimizer, str] = "Adam",
+    optimizer_parameters: dict | None = None,
+    lr_scheduler: Union[SupportedScheduler, str] = "ReduceLROnPlateau",
+    lr_scheduler_parameters: dict | None = None,
+) -> Union[FCNModule, VAEModule]:
+    """Create a CAREamics Lightning module.
+
+    This function exposes parameters used to create an AlgorithmModel instance,
+    triggering parameters validation.
+
+    Parameters
+    ----------
+    algorithm : SupportedAlgorithm or str
+        Algorithm to use for training (see SupportedAlgorithm).
+    loss : SupportedLoss or str
+        Loss function to use for training (see SupportedLoss).
+    architecture : SupportedArchitecture or str
+        Model architecture to use for training (see SupportedArchitecture).
+    algorithm_parameters : dict, optional
+        Algorithm parameters to use for training, by default {}. #TODO which ones?
```

Comment:
Here for instance, users will have absolutely no clue what parameters to put in this dictionary and will need to spend extra effort to figure it out (and it is not explain in the doc of this function).

Reply:
And I don't see where `algorithm_parameters` is used.